### PR TITLE
Fix/19 toggle button group

### PR DESCRIPTION
**Pull request by ektasawant**
Fixes [19](https://github.com/brightlayer-ui/angular-themes/issues/19) .

#### Changes proposed in this Pull Request:
- Added Outline variant in button toggle group
- Added Filled variant in button toggle group

#### Screenshots:
Screenshot 2022-01-13 at 11 29 15 AM
Screenshot 2022-01-13 at 11 38 17 AM

#### To Test:
- http://localhost:4200/material-components/input-components

### CODE REVIEW

**Comment by ektasawant** [comment]
@emclaug2 Do I need to raise PR in angular showcase demo repo for updating examples?

**Comment by huayunh** [comment]
Please double check the text color of the “outlined” variant and the “filled” variant.

**Comment by ektasawant** [comment]
> Please double check the text color of the “outlined” variant and the “filled” variant.

Done please review.
Screenshot 2022-01-13 at 5 14 53 PM

**Comment by emclaug2** [comment]
> @emclaug2 Do I need to raise PR in angular showcase demo repo for updating examples?

Yes this will require a separate PR to be made in the angular-showcase-demo repo.

File: mixins/button-toggle.scss [hunk 1]
```diff
@@ -0,0 +1,124 @@
+@import '../utility';
+
+@mixin blui-mat-button-toggle-default($palette) {
```
Comment (emclaug2) [inline mixins/button-toggle.scss:3]: I like the usage of mixins here but all three seem very similar to each other.  I'm looking at this now to see if we can simplify the logic a little bit.

**Comment by emclaug2** [comment]
Can you look at this proposed refactor and let me know your thoughts?  Since each set of style rules was the same except for the top level, I tweaked the mixin: 

https://github.com/brightlayer-ui/angular-themes/tree/feature/refactor-button-toggle-mixin

It should apply all the same rules as before.

**Comment by huayunh** [comment]
Visually looks good. I noticed a small shift as you click around, but I suppose that is beyond the control of plain CSS (unless one puts up some really tricky hacks with pseudo classes).

File: mixins/button-toggle.scss (hunk 1 above)
Comment (ektasawant) [reply mixins/button-toggle.scss]: Please review.

**Comment by ektasawant** [comment]
> 

Yes it causes because of `margin-left` CSS prop which I used for hiding double border. (I couldn’t find other solution). I will try to find solution with pseudo css.

**Comment by ektasawant** [comment]
> > Please double check the text color of the “outlined” variant and the “filled” variant.
> 
> Done please review. Screenshot 2022-01-13 at 5 14 53 PM

@huayunh , do we need hover color change as per variant change ? If yes can you guide me which figma design I can use?

**Comment by huayunh** [comment]
I think the default style is good enough for now.

**Comment by emclaug2** [comment]
This is going to require a little more work; for each button variant, the overall width of the button group changes by 1px dependent on which button is selected. Also the ripple animation overflows the container of the button.  Here's what I see: 

Strange

**Comment by jeffvg** [comment]
did we add another card that contains the variant types, I only have this one
http://localhost:4200/material-components/input-components

image

**Comment by ektasawant** [comment]
> did we add another card that contains the variant types, I only have this one http://localhost:4200/material-components/input-components
> 
> image

It is available in showcase demo repo https://github.com/brightlayer-ui/angular-showcase-demo/pull/46

**Comment by jeffvg** [comment]
ok didn't realize, looks good to me.

image

**Review by jeffvg — APPROVED** [review]

**Comment by ektasawant** [comment]
@jeffvg , need to fix mentioned issues https://github.com/brightlayer-ui/angular-themes/pull/58#issuecomment-1018679453. I will try to fix it tomorrow.

**Comment by ektasawant** [comment]
> This is going to require a little more work; for each button variant, the overall width of the button group changes by 1px dependent on which button is selected. Also the ripple animation overflows the container of the button. Here's what I see:
> 
> Strange
> 
> I'm not sure how to patch the changing width, but maybe a `overflow: hidden` on a nested div will solve the ripple overflow.

I have fixed ripple issue, but  changing width is difficult since using `margin-left: -1px`.

**Comment by huayunh** [comment]
Looking at this now. You would want to do a yarn update:submodule to pull in the changes in the `angular-showcase-demo` repository.

**Comment by ektasawant** [comment]
> yarn update:submodule
@huayunh , After running `yarn update:submodule` command do I need to push submodule file in this PR?
Screenshot 2022-01-28 at 7 33 22 PM

**Comment by huayunh** [comment]
I am by no means submodule authority (Joe is), but yes, yarn update:submodule should make your branch pointing to the latest commit in the showcase demo repo that you merged

**Comment by huayunh** [comment]
I am OK with merging this PR after the submodule pointer is updated since we got all the basic styles correct. We can log another enhancement to polish the math in the CSS.

**Comment by joebochill** [comment]
> > yarn update:submodule
> > @huayunh , After running `yarn update:submodule` command do I need to push submodule file in this PR? I am new to this process :( .
> 
> Screenshot 2022-01-28 at 7 33 22 PM

Yes you will need to push the pointer change in the submodule file

**Review by andrupu — COMMENTED** [review]
Aside from the submodule pointer, this looks great.

**Review by emclaug2 — APPROVED** [review]
@ektasawant Can you make a follow-up issue to address the 1px shift when toggling which button is active. I'll place an approval on this.

**Comment by emclaug2** [comment]
> This is going to require a little more work; for each button variant, the overall width of the button group changes by 1px dependent on which button is selected. Also the ripple animation overflows the container of the button. Here's what I see:
> 
> Strange
> 
> I'm not sure how to patch the changing width, but maybe a `overflow: hidden` on a nested div will solve the ripple overflow.



> @ektasawant Can you make a follow-up issue to address the 1px shift when toggling which button is active. I'll place an approval on this.

Nevermind I've opened the issue so we can merge this today.

**Pull request merged**